crates/bevy_ecs/src/event.rs

```diff
@@ -328,14 +328,25 @@ impl<T: Component> Events<T> {
         events.update();
     }
 
+    #[inline]
+    fn increment_start_event_count(&mut self) {
+        let count = self.event_count + 1;
+        self.a_start_event_count = count;
+        self.b_start_event_count = count;
```

Comment:
```suggestion
        self.a_start_event_count = self.event_count;
        self.b_start_event_count = self.event_count;
```

Just realized we can just use `event_count` directly. `event_count` corresponds to the "next" event, not the current event.

`event_count + 1` _probably_ also works due to how we reset event readers, but I'd rather not roll the dice :)